### ISSUE TITLE
Add support for more archive formats

### ISSUE THREAD

**Issue by monyarm** [libretro/RetroArch#4122]
I know RetroArch supports zip/7z files but i would like to request support for more archive formats, like for example zpaq http://mattmahoney.net/dc/zpaq.html

**Comment by hizzlekizzle** [comment]
Probably not going to happen. Integrating archive support is pretty difficult and we don't want to go around adding dependencies and support for every random archive format.

**Comment by monyarm** [comment]
could you atleast point me into the right direction for adding support for it for personal use?

**Comment by hizzlekizzle** [comment]
This is an old patch, but it's something like this:
https://github.com/libretro/RetroArch/pull/923

**Comment by meepingsnesroms** [comment]
I would be willing to try,but need a reason,zpaq is for archiving an entire filesystem multiple times not just one rom.

That 10GB benchmark is irrelevant since roms are single file and <1GB.

Using zfs I am using 56GB of a 40GB emulated harddrive,this is because thats where i store my source code,and lots of source code overlaps.(like 3 ndk versions don't need 3 copys of stdio.h)

Do you know of a single file compression algorithm more efficient than 7z?

**Comment by monyarm** [comment]
@meepingsnesroms i know the benchmark is irrevelant for rom files (plus the extraction time for a file that large would be ridiculous) and yes i do know a single file compression algorithm more efficient than 7z, zpaq, i've tried compressing many different rom formats, and have found incredible savings compared to 7z (well depenging on the rom format), for my gba collection i've saved a hundred megabytes compared to 7z, and for my n64 collection a gigabyte, and have saved gigabyes for my entire rom collection compared to 7z, sure zpaq is great for entire filesystems and has a benchmark for that, but most of the paq variants (that zpaq is based on) are made for compressing a single filetype with ridiculuous compression ratios (as in a text file going from 500 to 50 megabytes) what zpaq does is aggregate all of the variants (or at least most of them)

**Comment by meepingsnesroms** [comment]
Are all those ROMs in 1 zpaq archive,or each in there own?

**Comment by monyarm** [comment]
@meepingsnesroms i make them individual unless they are variants (think pokemon fire red and leaf green) though putting them in 1 archive should work too.

**Comment by meepingsnesroms** [comment]
Could you list the archive formats you want.(since you refer to multiple but your only example is zpaq)

Is rar supported yet?

**Comment by monyarm** [comment]
i don't believe retroarch supports rar, as for a list of file formats

zpaq
rar
and those are the ones i can think of off the top of my head, though since rar has similiar if not even worse savings than 7z i don't really see a reason to support it

**Comment by retro-wertz** [comment]
imagine this supported for romsets like fba or mame, thats gonna ba a loooong list of crc to track not to mention the method these roms are scanned and added to playlist. is there something like torrentzpaq?

**Comment by monyarm** [comment]
what do you mean by "torrentzpaq" ? And honestly im simply asking for this cause i want to fit as many roms as possible on my usb drive, and zpaq gives much better savings compared to 7z.

**Comment by meepingsnesroms** [comment]
I think there asking if zpaq is a torrent alternative?

**Comment by cudencuden** [comment]
there's also zstandard... better compression n faster uncompression. https://github.com/facebook/zstd

**Comment by Ferk** [comment]
https://sourceforge.net/projects/trrntzip/
> TorrentZip is a replacement for MameZip. The goal of the program is to use standard values when creating zips to create identical files over multiple systems. 

For what I understand, ZPAQ can store a SHA-1 checksum of the original data that can be checked without uncompressing the whole thing. This might make it actually more suitable than zips when you want to identify the content by checksum.

**Comment by monyarm** [comment]
Yes zpaq does store checksums, it was added in version 0.06, while the current version is 7.15
